### PR TITLE
feat(chat): Add download to Application logs (Issue #2581)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

Add download button to the logs window

Issues:

- Issue #2581 

**UI changes**

Снимок экрана 2024-11-12 в 15 06 13
Снимок экрана 2024-11-12 в 15 05 56

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Derikyan** [comment]
deploy-review

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2588.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 1]
```diff
@@ -62,27 +50,32 @@ const LogsView = ({ applicationLogs }: LogsViewProps) => {
...
+  applicationLogs,
+  isLogsLoading,
+}: LogsFooterProps) => {
   const { t } = useTranslation(Translation.Marketplace);
   const dispatch = useAppDispatch();
 
-  const isLogsLoading = useAppSelector(
-    ApplicationSelectors.selectIsLogsLoading,
-  );
+  const handleDownload = async (applicationLogs: string) => {
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]:
```suggestion
  const handleDownload = (applicationLogs: string) => {
```

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 2]
```diff
@@ -103,6 +96,21 @@ const LogsFooter = ({ entityId }: LogsFooterProps) => {
           />
         </button>
       </Tooltip>
+      {applicationLogs && (
+        <Tooltip tooltip={t('Download logs')}>
+          <button
+            onClick={() => handleDownload(applicationLogs)}
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: you can call `downloadApplicationLogs(applicationLogs)` here. `handleDownload`. handleDownload is redundant I guess

File: apps/chat/src/utils/app/import-export.ts [hunk 3]
```diff
@@ -175,6 +175,19 @@ function downloadChatPromptData(
   );
 }
 
+export function downloadApplicationLogs(data: string, fileName?: string) {
+  const baseName = fileName
+    ? `${fileName}_application_logs`
+    : 'application_logs';
+
+  const exportedFileName = `${baseName}_${currentDate()}.txt`;
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/import-export.ts]:
```suggestion
const exportedFileName = [fileName, 'application_logs', currentDate()].filter(Boolean).join('_');
```

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 4]
```diff
@@ -132,8 +143,15 @@ export const ApplicationLogs = ({
       onClose={onClose}
     >
       <LogsHeader />
-      <LogsView applicationLogs={applicationLogs} />
-      <LogsFooter applicationLogs={applicationLogs} entityId={entityId} />
+      <LogsView
+        applicationLogs={applicationLogs}
+        isLogsLoading={isLogsLoading}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: `LogsView` and `LogsFooter` can read these data directly from state

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2588.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2588.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 5]
```diff
@@ -103,6 +91,26 @@ const LogsFooter = ({ entityId }: LogsFooterProps) => {
...
+          <button
+            onClick={() => downloadApplicationLogs(applicationLogs)}
+            className="button button-secondary flex h-[38px] items-center gap-1"
+            data-qa="application-download-logs"
+            disabled={isLogsLoading}
+          >
+            <IconDownload
+              className={classNames(
+                isLogsLoading
+                  ? 'cursor-not-allowed bg-layer-3 text-controls-disable'
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: it's already added by you into `.button.button-secondary:disabled` in `globals.css`

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2588.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**